src/main/java/de/dhbw/karlsruhe/models/Grammar.java

```diff
@@ -1,12 +1,22 @@
 package de.dhbw.karlsruhe.models;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class Grammar {
 
   private String[] terminals;
   private String[] nonTerminals;
   private String[] productions;
   private String startSymbol;
 
+  public Grammar(String[] pTerminals, String[] pNonTerminals, List<GrammarRule> pGrammarRules, String pStartSymbol) {
```

Comment:
Warum pTerminals & pNonTerminals usw,?

Reply:
Habe mal gelernt/gehört, dass man übergebene Parameter immer mit einem p kennzeichnet also quasi pParamName. Mir ist aber auch schon aufgefallen, dass es nicht immer in jeder Klasse 100% einheitlich ist. Dennoch glaube ich, dass es nicht schlecht ist

Reply:
Dadurch kann es auch zu keinen Verwechslungen kommen mit anderen Variablen der Klassen, auch wenn man darauf natürlich normalerweise immer achten sollte.